### PR TITLE
Document K8s CRDs and COs

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: content/working_with/official_plugins/Orchestration/kubernetes.md [hunk 1]
```diff
@@ -718,3 +719,105 @@ and now, using the kubernetes plugin it creates resource in the cluster(pod):
         target: sanity_master
 
 {{< /highlight >}}
+
+
+## CustomResourceDefinition and Custom Objects
+
+The Cloudify Kubernetes Plugin supports creating and using [CustomResourceDefinition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/)s.
```
Comment (AdarShaked) [inline content/working_with/official_plugins/Orchestration/kubernetes.md:726]: [CustomResourceDefinitions] ?

**Review by AdarShaked — APPROVED** [review]

File: content/working_with/official_plugins/Orchestration/kubernetes.md (hunk 1 above)
Comment (EarthmanT) [reply content/working_with/official_plugins/Orchestration/kubernetes.md:726]: No

Comment (ofercloudify) [inline content/working_with/official_plugins/Orchestration/kubernetes.md:726]:
```suggestion
The {{< param product_name >}} Kubernetes Plugin supports creating and using [CustomResourceDefinition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/)s.
```

File: content/working_with/official_plugins/Orchestration/kubernetes.md [hunk 2]
```diff
@@ -718,3 +719,105 @@ and now, using the kubernetes plugin it creates resource in the cluster(pod):
...
+      file:
+        resource_path: resources/crd.yaml
+```
+
+In your `crd.yaml` file, provide the definition for your **CustomDefinedResource**.
+
+You can also use **CustomDefinedResource**s with the Kubernetes plugin. Usages of **CustomDefinedResource**s are referred to as [Custom Objects](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-custom-objects).
+When using **Custom Objects**, you must add specific `annotations` to the `metadata` section of the resource.
+Specify the following keys in your `metadata:annotations` section: `['cloudify-crd-group', 'cloudify-crd-plural', 'cloudify-crd-version']`.
+Cloudify needs these keys in order to identify the resource as a usage of a **CustomDefinedResource**.
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Orchestration/kubernetes.md:744]:
```suggestion
{{< param product_name >}} needs these keys in order to identify the resource as a usage of a **CustomDefinedResource**.
```

**Review by ofercloudify — CHANGES_REQUESTED** [review]

**Pull request closed without merging**